sympy/matrices/common.py

```diff
@@ -2377,17 +2397,21 @@ def __pow__(self, exp):
 
         return self.pow(exp)
 
-    def pow(self, exp, jordan=None, dotprodsimp=None):
+
+    def pow(self, exp, method=None, dotprodsimp=None):
+
         """Return self**exp a scalar or symbol.
 
         Parameters
         ==========
 
-        jordan : bool, optional
-            If left as None then Jordan form exponentiation will be used under
-            certain conditions, True specifies that jordan_pow should always
-            be used if possible and False means it should not be used unless
-            it is the only way to calculate the power.
+
+        method : jordon, cayley
```

Comment:
It should also be possible to specify recursion

Reply:
Perhaps recursion should be specified as "multiply" though since I think that makes more sense to users.

Reply:
Actually since dotprodsimp is only used for recursion it should probably be:
```
        method : multiply, mulsimp, jordon, cayley
```
Then mulsimp corresponds to using recursion with dotprodsimp and the dotprodsimp parameter can be removed (it was added since last release).

Reply:
I have made the changes. Is is good now?